src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -278,39 +279,41 @@ def is_missing(self, timestamp: datetime) -> bool:
         return any(map(lambda gap: gap.contains(timestamp), self._gaps))
 
     def _update_gaps(
-        self, timestamp: datetime, newest: datetime, new_missing: bool
+        self, timestamp: datetime, newest: datetime, should_be_in_gaps: bool
     ) -> None:
         """Update gap list with new timestamp.
 
         Args:
             timestamp: Timestamp of the new value.
             newest: Timestamp of the newest value before the current update.
-            new_missing: if true, the given timestamp will be recorded as missing.
-
+            should_be_in_gaps: if true, the given timestamp will be recorded as missing.
         """
-        currently_missing = self.is_missing(timestamp)
+        found_in_gaps = self.is_missing(timestamp)
 
-        if not new_missing:
+        if not should_be_in_gaps:
             # Replace all gaps with one if we went far into then future
-            if self._datetime_newest - newest >= self._time_range:
+            if self._datetime_newest - newest >= self._sampling_period * len(
+                self._buffer
+            ):
```

Comment:
So the bug fix in the first commit is here, right? Two comments:

1. In terms of commit, maybe it would have made much more obvious where the bug fix was if you split the commit in removing `self._time_range` first (which still has the bug) and then a second commit just fixing the bug.
2. Would it be too complicated to add a test that fails before the bug fix and succeeds after to make sure this doesn't regress in the future?

Reply:
I mean, we have a test like that, the added test will fail if this logic fails

Reply:
And why didn't tests failed before? I'm lost.

Reply:
Basically this PR fixes two bugs:
* In __len__() we forgot to consider gaps, that was added and a test for it as well.
* For specific situations, the gaps were wrong which the bug in __len__() exposed and thus is also tested in the same tests as for __len__() 

But after your comments I added an extra test for the second (gap) bug as well now